server/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -97,7 +98,7 @@ public final String simpleName() {
      *         users configure synthetic source in the mapping without configuring all
      *         fields properly.
      */
-    public SourceLoader.SyntheticFieldLoader syntheticFieldLoader() {
+    public SourceLoader.SyntheticFieldLoader syntheticFieldLoader(NestedScope nestedScope) {
```

Comment:
Iiuc, this is only used in object mappers? If so, maybe we add a second flavor for use in these mappers or move this version there. Would something like this work?

Reply:
I reverted the `NestedScope` change and adjusted the nested builder/context to get the information when the mapping is built:
https://github.com/elastic/elasticsearch/pull/109117/commits/315d3c4a5bf72ad537c4b7a0dbb5c57ef42ae33e

Reply:
Nice, it's much simpler now.. It seems reasonable to me, I'd say we continue with breaking it to smaller PRs to wrap this up.